crates/weaver_resolver/src/registry.rs

```diff
@@ -685,6 +694,40 @@ fn resolve_inheritance_attr(
     }
 }
 
+fn resolve_body(
+    ureg: &mut UnresolvedRegistry,
+    attr_catalog: &mut AttributeCatalog,
+) -> Result<(), Error> {
+    let mut errors = vec![];
+
+    for unresolved_group in ureg.groups.iter_mut() {
+        let body = &unresolved_group.body;
```

Comment:
Nit - this can be cleaned by using rust's if-let syntax.

```rust
if let Some(body) = &unresolved_group.body {
  match resolve_body_spec(&body.spec, attr_catalog) {
     Ok(resolved_body) => {unresolved_group.group.body = resolved_body;},
     Err(e) = { errors.push(...) }
  }
}
// No need for `None` case
```

Second nit: You may want to preserve more than just an error string from `resolve_body_spec` failure.

Reply:
Changed to "keep" the error -- not sure this is exactly what we want but lets review.